pkg/syncer/fullsync.go

```diff
@@ -630,6 +655,57 @@ func patchVolumeAccessibleTopologyToPVC(ctx context.Context, k8sClient clientset
 	return nil
 }
 
+func setFileShareAnnotationsOnPVC(ctx context.Context, k8sClient clientset.Interface,
+	volumeManager volumes.Manager, pvc *v1.PersistentVolumeClaim) error {
+	log := logger.GetLogger(ctx)
+	log.Infof("setFileShareAnnotationsOnPVC: Setting file share annotation for pvc: %q, namespace: %q",
+		pvc.Name, pvc.Namespace)
+
+	pv, err := k8sClient.CoreV1().PersistentVolumes().Get(ctx, pvc.Spec.VolumeName, metav1.GetOptions{})
```

Comment:
I think we have a cache which has PVC to volumeID mapping.

https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go#L2019

Reply:
Your code changes are only relevant when WCP cluster also gets upgraded - that is CSI driver is upgraded. If CSI driver is old, your code will not be there to begin with.

Reply:
Updated code to fetch volumeID for PVC from cache.

Reply:
Found that this change didn't work and it was unable to get volumeID for PVC from cache. Hence reverted the change to original code.